ast/Trees.h

```diff
@@ -783,6 +784,25 @@ TREE(Array) : public Expression {
 };
 CheckSize(Array, 56, 8);
 
+TREE(Range) : public Expression {
+public:
+    TreePtr from;
+    TreePtr to;
+    bool exclusive;
+
+    Range(core::LocOffsets loc, TreePtr from, TreePtr to, bool exclusive);
+
+    TreePtr deepCopy() const;
+
+    virtual std::string toStringWithTabs(const core::GlobalState &gs, int tabs = 0) const;
+    virtual std::string showRaw(const core::GlobalState &gs, int tabs = 0);
+    virtual std::string nodeName();
+
+private:
+    virtual void _sanityCheck();
+};
+CheckSize(Range, 56, 8);
+
```

Comment:
I don't think you need the Range tree at all: you've defined the `buildRange` intrinsic on the `<Magic>` symbol already, and should just translate range syntax directly into that intrinsic in Desugar.cc.

Adding a new tree node is a really heavyweight solution, and will touch most passes. It would be great to avoid this if possible.